File: ios/VideoApp/Sources/Models/Notification.swift
import Foundation

struct NotificationItem: Decodable, Identifiable {
    let id: String
    let type: String?
    let content: String?
    let fromUserId: String?
    let toUserId: String?
    let relatedVideoId: String?
    let relatedCommentId: String?
    let read: Bool
    let createTime: String?
}

struct NotificationListResponse: Decodable {
    let list: [NotificationItem]
    let total: Int
    let totalPages: Int
    let currentPage: Int
    let pageSize: Int
}

struct MarkAsReadRequest: Encodable {
    let notificationId: String
}

File: android/app/src/main/java/com/github/makewheels/video2022/data/model/Notification.kt
package com.github.makewheels.video2022.data.model

import java.util.Date

data class NotificationItem(
    val id: String,
    val type: String?,
    val content: String?,
    val fromUserId: String?,
    val toUserId: String?,
    val relatedVideoId: String?,
    val relatedCommentId: String?,
    val read: Boolean = false,
    val createTime: String?
)

data class NotificationListResponse(
    val list: List<NotificationItem>,
    val total: Long,
    val totalPages: Int,
    val currentPage: Int,
    val pageSize: Int
)
